code/modules/projectiles/guns/projectile/potion_launcher.dm

```diff
@@ -0,0 +1,65 @@
+/obj/item/weapon/gun/potionlauncher
+	name = "potion launcher"
+	icon = 'icons/obj/gun.dmi'
+	icon_state = "potionlauncher"
+	item_state = "potionlauncher"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guninhands_left.dmi', "right_hand" = 'icons/mob/in-hand/right/guninhands_right.dmi')
+	desc = "The Wizard Federation's answer to the ongoing arms race with NanoTrasen and the Syndicate. This launcher can hold up to 5 potions for rapid potion throwing."
+	w_class = W_CLASS_LARGE
+	throw_speed = 2
+	throw_range = 10
+	force = 5.0
+	var/list/potions = new/list()
+	var/max_potions = 5
+	starting_materials = list(MAT_IRON = 2000)
+	w_type = RECYK_METAL
+
+/obj/item/weapon/gun/potionlauncher/examine(mob/user)
+	..()
+	if(!(potions.len))
+		to_chat(user, "<span class='info'>It is empty.</span>")
+		return
+	to_chat(user, "<span class='info'>It has [potions.len] / [max_potions] potions loaded.</span>")
+	for(var/obj/item/potion/G in potions)
+		to_chat(user, "[bicon(G)] [G.name]")
+
+/obj/item/weapon/gun/potionlauncher/attackby(obj/item/I as obj, mob/user as mob)
+
+	if((istype(I, /obj/item/potion)))
+		if(potions.len < max_potions)
+			if(user.drop_item(I, src))
+				potions += I
+				to_chat(user, "<span class='notice'>You load the [I.name] into the [src.name].</span>")
+				to_chat(user, "<span class='notice'>[potions.len] / [max_potions] potions loaded.</span>")
+		else
+			to_chat(user, "<span class='warning'>The [src.name] cannot hold more potions.</span>")
+
```

Comment:
May want to return 1 if the item is a potion, else the potions will call afterattack, which may cause you to have a flaming, sword-impaled, potion launcher